src/recommenders/base.py

```diff
@@ -1,25 +1,63 @@
 from abc import ABC, abstractmethod
 
+import json
 import numpy as np
 import pandas as pd
 
 from src.configs import get_config
 
 
 class BaseRecommender(ABC):
-    def __init__(self, stats_df: pd.DataFrame):
+    def __init__(self, stats_df: pd.DataFrame, player_name):
         self.stats_df = stats_df
         self.categories = self.stats_df.columns[1:5]
         self.cat_ranges = [np.unique(self.stats_df[col]) for col in self.categories]
+        self.score = None
+        self.config = None
+        self.optimizer_name = None
+        self.player_name = player_name
 
     @abstractmethod
-    def recommend(self):
+    def __recommend(self, player_name):
         raise NotImplementedError
 
+    def recommend(self):
+        self.__recommend()
+        self.dump_to('output.json', ['config', 'optimizer_name', 'score', ])
+
     def _objective(self, *args):
+        print(args)
         config = get_config(pd.Series(args[0], index=self.categories).T)
+        self.optimizer_name = 'bo'
+        self.config = config
         print()
         print(f'Config: {config}')
         score = int(input('Enter score: '))
         print()
+        self.score = score
         return score
+
+    def dump_to(self, filename, var_names):
```

Comment:
Follow [PEP8](http://www.python.org/dev/peps/pep-0008/): This should be named `_dump_to`.

> `_single_leading_underscore`: weak "internal use" indicator. E.g. `from M import *` does not import objects whose name starts with an underscore.

Reply:
@korchiev Please rename.